tests/nodes/tests_Node.cpp

```diff
@@ -222,6 +222,31 @@ TEST_CASE("IRI UUID") {
     CHECK(regex::Regex{"^urn:uuid:[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$"}.regex_match(uuid.identifier()));
 }
 
+TEST_CASE("IRI fetch or serialize") {
+    auto const x = IRI::make("https://example.com#some-iri");
+    auto const expected = x.identifier();
+
+    std::string buf;
+    writer::StringWriter w{buf};
+
+    std::string_view fetched;
+    switch (x.fetch_or_serialize_identifier(fetched, w)) {
```

Comment:
No need for a switch-cases here. 

```
CHECK_EQ(x.fetch_or_serialize_identifier(fetched, w), FetchOrSerializeResult::Fetched)
```

will do because IRI cannot inline. So, it will not serialize.